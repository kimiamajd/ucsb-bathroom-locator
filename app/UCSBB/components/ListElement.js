import React from 'react';

import Colors from '../constants/Colors';
import InfoIcon from '../components/InfoIcon';
import { Text, View } from 'react-native';


export default function ListElement(props) {
  var whitespace = '                                                     ';
    return (
        <Text style = {{ fontSize: 24, fontFamily: 'Courier', textAlign: 'left' }}> 
        {(props.text + whitespace).slice(0,21)} 
        {props.access=='wheelchair' ? <InfoIcon name = 'wheelchair' /> : <></>}
        {' '}
        <InfoIcon name = {props.gender} />
        </Text>
      );
    
}
