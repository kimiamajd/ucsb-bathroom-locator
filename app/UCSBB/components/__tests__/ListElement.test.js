import React from 'react';
import {shallow} from 'enzyme';
import ListElement from '../ListElement';


describe('List Element', () => {
    describe('Rendering Male/Accessible', () => {
        it('should match to snapshot', () => {
            const component = shallow(<ListElement text='Test 0001' gender='male' access='wheelchair'/>)
            expect(component).toMatchSnapshot()
        });
    });
});

describe('List Element', () => {
    describe('Rendering Male/Not Accessible', () => {
        it('should match to snapshot', () => {
            const component = shallow(<ListElement text='Test 0002' gender='male' access='none'/>)
            expect(component).toMatchSnapshot()
        });
    });
});