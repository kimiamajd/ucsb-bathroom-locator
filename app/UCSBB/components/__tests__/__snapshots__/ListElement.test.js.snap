// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Element Rendering Female/Accessible should match to snapshot 1`] = `ShallowWrapper {}`;

exports[`List Element Rendering Female/Not Accessible should match to snapshot 1`] = `ShallowWrapper {}`;

exports[`List Element Rendering Male/Accessible should match to snapshot 1`] = `ShallowWrapper {}`;

exports[`List Element Rendering Male/Not Accessible should match to snapshot 1`] = `ShallowWrapper {}`;

exports[`List Element Rendering Neutral/Accessible should match to snapshot 1`] = `ShallowWrapper {}`;

exports[`List Element Rendering Neutral/Not Accessible should match to snapshot 1`] = `ShallowWrapper {}`;
